### PR TITLE
Filter script

### PR DESCRIPTION
**Pull request by moltubaer**

### CODE REVIEW

**Comment by nemisis84** [comment]
I suggest we make it a class so that it easier to pass into other files. One alternative could be to make a main method

File: scripts/filter.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    # Splits a log file into different logs based on the Log ID
+    for ID in logIDArr:
+        open(logDirPath + ID + ".log", "w").writelines(line for line in open(input) if ID in line)
+    os.remove(input)
+
+
+def formatFiles(inputArr):
+    # Formats the file into a more readable form
+    for ID in inputArr:
+        with open(logDirPath + ID + '.log', 'r') as file:
```
Comment (nemisis84) [inline scripts/filter.py:36]: Give the open both read and write permissions when first opening it. We don't need to open it twice

**Review by nemisis84 — CHANGES_REQUESTED** [review]
Ser bra ut med noen småting. Jeg foretrekker OOP tilnærming

File: scripts/filter.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
+import os
+
+logDirPath = '../logfiles/'
+logFile = logDirPath + 'adapter_multi_call.log'
+# logFile = logDirPath + 'adapter_single_call.log'
+temp = logDirPath + 'temp.log'
+outputFile = logDirPath + 'output.log'
+
+filterDictionary = {
+    "fields": ["SIP/2.0", "From", "To", "Via", "Contact", "Call-ID", "CSeq", "Content-Length", "Max-Forwards", "Content-Type", "Route", "Record-Route"],
```
[the remark this answers is not in the document]
Comment (moltubaer) [reply scripts/filter.py:10]: It is hard to filter out "Record-Route" without interfering with the filtering of "Route", with the current algorithm. I suggest leaving it as is.

File: scripts/filter.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    "codec": ["a=", "b=", "c=", "i=", "k=", "m=", "o=", "r=", "s=", "t=", "v=", "z=",],
+}
+logIDArr = []
+
+def isolateMessages(file, output):
+    # Isolates message packets
+    open(output, "w").writelines(line for line in open(file) if 'message with id' in line)
+    # Identifies traces by log ID and adds them into an array
+    with open(output) as file:
```
[the remark this answers is not in the document]
Comment (moltubaer) [reply scripts/filter.py:19]: I have to open it once for writing and a second time for reading. I have tried combining it with a "read/write" mode, but it has so far proven unsuccessful. It takes too much time for it to be worth it.

**Review by nemisis84 — APPROVED** [review]

**Pull request merged**